wallet/src/account/output_cache/mod.rs

```diff
@@ -908,13 +948,14 @@ impl OutputCache {
             match input {
                 TxInput::Utxo(outpoint) => {
                     self.consumed.insert(outpoint.clone(), tx.state());
-                    if is_unconfirmed {
-                        self.unconfirmed_descendants
-                            .get_mut(&outpoint.source_id())
-                            .as_mut()
-                            .map(|descendants| descendants.insert(tx_id.clone()));
-                    } else {
-                        self.unconfirmed_descendants.remove(tx_id);
+                    if let Some(descendants) =
+                        self.unconfirmed_descendants.get_mut(&outpoint.source_id())
+                    {
+                        if is_unconfirmed {
+                            descendants.insert(tx_id.clone());
+                        } else {
+                            descendants.remove(tx_id);
+                        }
```

Comment:
It doesn't make sense if the tx is confirmed for it to have a utxo pointing to an unconfirmed tx. Maybe error out in the else instead of descentents.remove ?